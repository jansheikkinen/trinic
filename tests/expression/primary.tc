// Primary

function main(void) -> void do
  (1 + 2); 1; -1; 1.1; "string"; identifier; true; false; { 1, 2, 3 };
  function(void) -> void do x; end
end
