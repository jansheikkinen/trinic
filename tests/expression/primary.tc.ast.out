## PARSER ##
Parsing tests/expression/primary.tc...

### LEXER ###
Generating tokens from source...
Generated 48 tokens
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:            (,     (null), (004, 003)
009:      INTEGER,          1, (004, 005)
010:            +,     (null), (004, 006)
011:      INTEGER,          2, (004, 009)
012:            ),     (null), (004, 009)
013:            ;,     (null), (004, 010)
014:      INTEGER,          1, (004, 013)
015:            ;,     (null), (004, 013)
016:            -,     (null), (004, 015)
017:      INTEGER,          1, (004, 017)
018:            ;,     (null), (004, 017)
019:        FLOAT,        1.1, (004, 022)
020:            ;,     (null), (004, 022)
021:       STRING,     string, (004, 030)
022:            ;,     (null), (004, 031)
023:   IDENTIFIER, identifier, (004, 042)
024:            ;,     (null), (004, 043)
025:         TRUE,     (null), (004, 048)
026:            ;,     (null), (004, 049)
027:        FALSE,     (null), (004, 055)
028:            ;,     (null), (004, 056)
029:            {,     (null), (004, 058)
030:      INTEGER,          1, (004, 061)
031:            ,,     (null), (004, 061)
032:      INTEGER,          2, (004, 064)
033:            ,,     (null), (004, 064)
034:      INTEGER,          3, (004, 067)
035:            },     (null), (004, 068)
036:            ;,     (null), (004, 069)
037:     FUNCTION,     (null), (005, 010)
038:            (,     (null), (005, 011)
039:         VOID,     (null), (005, 015)
040:            ),     (null), (005, 016)
041:           ->,     (null), (005, 018)
042:         VOID,     (null), (005, 023)
043:           DO,     (null), (005, 026)
044:   IDENTIFIER,          x, (005, 028)
045:            ;,     (null), (005, 029)
046:          END,     (null), (005, 033)
047:          END,     (null), (006, 003)

### AST GENERATION ###
function main(VOID) -> VOID do
  ((1 + 2))
  1
  -1
  1.100000
  "string"
  identifier
  true
  false
  {1, 2, 3,   }
  function (VOID) -> VOID do
  x
  end


  end

